### PR TITLE
Fix : Reduce Codacy issues wherever possible

### PR DESCRIPTION
**Pull request by amitiwary999**
Fixes issue #682 

Changes: some changes as mentioned [here](https://www.codacy.com/app/harshithdwivedi/susi_android/issues?step=0&filters=W3siaWQiOiJDYXRlZ29yeSIsInZhbHVlcyI6WyJFcnJvciBQcm9uZSJdfV0%3D)

### CODE REVIEW

**Comment by chiragw15** [comment]
@amitiwary999 @mayank408 can we a separate helper class for dialog box and add these methods to it? Then it can be reused in any activity.

**Comment by amitiwary999** [comment]
Each alert box message is different and also action of button is different that's why I use different method for each alert box.

**Comment by chiragw15** [comment]
Yes, different methods are fine, but these methods are repeated in different activities. Got my point? Like see alertException(Throwable t) is repeated in Login, Signup and ForgotPassword activity.

**Comment by amitiwary999** [comment]
@chiragw15 So you want me to add helper class for only exceptionalert?

**Comment by chiragw15** [comment]
No, for all dialog alert boxes.

**Comment by chiragw15** [comment]
If it's feasible. I guess, it should be.

**Comment by amitiwary999** [comment]
You mean implement all alert dialog box in helper class and exception alert is almost common so only single alert for it.

**Comment by chiragw15** [comment]
I think only one dialog box can also solve the problem. You can use a callback for onclick listeners. Not sure though. Try it once. The main aim is to minimize the code. So, remove redundancy as much as you can.

**Comment by amitiwary999** [comment]
@chiragw15 @mayank408 review please.

**Comment by amitiwary999** [comment]
@chiragw15 @mayank408 please review. After merging development branch circleci test fail.

File: app/src/main/java/org/fossasia/susi/ai/activities/ForgotPasswordActivity.java [hunk 1]
```diff
@@ -202,4 +161,45 @@ protected void onSaveInstanceState(Bundle outState) {
...
+                finish();
+            }
+        };
+        alertTitle = null;
+        alertMessage = response.body().getMessage();
+        AlertboxHelper successAlertboxHelper = new AlertboxHelper(ForgotPasswordActivity.this,alertTitle,alertMessage,dialogClickListener,null,"Continue",null,Color.BLUE);
+        successAlertboxHelper.showAlertBox();
+    }
+
+    public void alertFailure() {
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/activities/ForgotPasswordActivity.java]: First thing, alertFailure() , alertError() and alertException can be combined. Send title and message as parameters. Secondly, see these three functions are exactly same in login activity and forgot password activity. So, can't we just move that code in util and reuse it.

Comment (amitiwary999) [inline app/src/main/java/org/fossasia/susi/ai/activities/ForgotPasswordActivity.java]: which three function alertFailure(),alertError(),alertException()?

Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/activities/ForgotPasswordActivity.java]: Just combine these functions alertFailure(),alertError(),alertException(). Single method can replace these 3. Pass 'title' and 'message' as parameters. Forgot about second point for now. Will see that in future.

File: app/src/main/java/org/fossasia/susi/ai/activities/ForgotPasswordActivity.java [hunk 2]
```diff
@@ -202,4 +169,24 @@ protected void onSaveInstanceState(Bundle outState) {
...
+    public void alertSuccess(Response<ForgotPasswordResponse> response) {
+        DialogInterface.OnClickListener dialogClickListener = new DialogInterface.OnClickListener() {
+            public void onClick(DialogInterface dialog, int id) {
+                Intent intent = new Intent(ForgotPasswordActivity.this, LoginActivity.class);
+                intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
+                startActivity(intent);
+                finish();
+            }
+        };
+        alertTitle = null;
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/activities/ForgotPasswordActivity.java]: Don't keep title of dialog box null

File: app/src/main/java/org/fossasia/susi/ai/activities/ForgotPasswordActivity.java [hunk 3]
```diff
@@ -129,43 +131,15 @@ public void onCancel(DialogInterface dialog) {
...
-                    AlertDialog.Builder builder = new AlertDialog.Builder(ForgotPasswordActivity.this);
-                    builder.setTitle(response.code() + " Error");
-                    builder.setMessage(response.message())
-                            .setCancelable(false)
-                            .setPositiveButton("OK", null);
-                    AlertDialog alert = builder.create();
-                    alert.show();
-                    Button pbutton = alert.getButton(DialogInterface.BUTTON_POSITIVE);
-                    pbutton.setTextColor(Color.BLUE);
+                    alertNotSuccess(response.code()+" Error",response.message(),"OK",Color.BLUE);
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/activities/ForgotPasswordActivity.java]: Use spaces between parameters to make it look neat. Same everywhere.

File: app/src/main/java/org/fossasia/susi/ai/activities/SignUpActivity.java [hunk 4]
```diff
@@ -281,4 +226,57 @@ public void onClick(DialogInterface dialogInterface, int i) {
...
+            }
+        };
+        alertTitle = getResources().getString(R.string.error_email);
+        alertMessage = getResources().getString(R.string.error_msg);
+        AlertboxHelper failureAlertboxHelper = new AlertboxHelper(SignUpActivity.this,alertTitle,alertMessage,dialogClickListener,dialogClickListenern,"OK","Forgot Password",getResources().getColor(R.color.md_blue_500));
+        failureAlertboxHelper.showAlertBox();
+
+    }
+
+    public void alertException(Throwable t) {
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/activities/SignUpActivity.java]: alertException() and alertError() can be combined

**Comment by chiragw15** [comment]
@amitiwary999 please do the changes requested.

**Comment by amitiwary999** [comment]
I will change it by tonight.

**Comment by amitiwary999** [comment]
@chiragw15 updated code.

File: app/src/main/java/org/fossasia/susi/ai/activities/ForgotPasswordActivity.java (hunk 3 above)
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/activities/ForgotPasswordActivity.java]: @amitiwary999 this ^

**Pull request closed without merging**